### PR TITLE
WIP: Add name allowlist to restrict intermediate step streaming

### PR DESCRIPTION
**Pull request by jacoblee93**

### CODE REVIEW

File: langserve/client.py [hunk 1]
```diff
@@ -455,6 +459,8 @@ async def astream_log(
             async with aconnect_sse(
                 self.async_client, "POST", endpoint, json=data
             ) as event_source:
+                event_source.response.raise_for_status()
```
Comment (efriis) [inline langserve/client.py:462]: How do we want to handle http exceptions in the streaming endpoints? If there's a way to split this request into sync (for exceptions) and async (for streaming response) that would be ideal, so we can read the response error message

File: tests/unit_tests/test_server_client.py [hunk 2]
```diff
@@ -349,6 +349,109 @@ async def test_astream_log(async_client: RemoteRunnable) -> None:
...
+    )
+    add_routes(
+        app,
+        RunnableLambda(add_one),
+        input_type=int,
+        path="/allowlist",
+        stream_log_name_allowlist=["allowed"],
+    )
+
+    async with get_async_client(app, path="/empty_allowlist") as runnable:
```
Comment (efriis) [inline tests/unit_tests/test_server_client.py:376]: The fact that this didn't end in a slash triggered the client error. Might be worth adding a test specifically for that.

File: langserve/client.py (hunk 1 above)
Comment (nfcampos) [reply langserve/client.py:461]: An error event in the stream

**Comment by eyurtsev** [comment]
Already implemented

**Pull request closed without merging**